### PR TITLE
Separate the hash calculation from the IRs renaming

### PR DESCRIPTION
**Pull request by andrei-kochin**

### CODE REVIEW

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import defusedxml.ElementTree as ET
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py]: we can remove xml related imports

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        env = set_env_variable(env, LD_LIB_PATH_NAME, ov_bin_path)
+        logger.warning("Set the following env varibles to calculate the IR-based hash: ")
+        logger.warning(f'PYTHONPATH={env["PYTHONPATH"]}')
+        logger.warning(f'{LD_LIB_PATH_NAME}={env[LD_LIB_PATH_NAME]}')
+        exit(0)
+    else:
+        logger.error(f'Impossible to run the tool! PyOpenVINO was not found!')
+        exit(-1)
+
+XML_EXTENSION = ".xml"
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py:37]: Should be in `utils/constant.py`

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+BIN_EXTENSION = ".bin"
+
+@dataclass
+class TestStructure:
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py:41]: can remove this class

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                        pass
+            except:
+                logger.error(f"Impossible to create hash for the{model_path}")
+
+            if not str_to_hash:
+                hash = "INVALID"
+            else:
+                hash = str(sha256(str_to_hash.encode('utf-8')).hexdigest())
+
+            logger.info(f"Generated hash for the {model_path}: {hash}")
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py:158]: Not rename?

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+            shape_str += str(partial_shape.is_dynamic)
+        except:
+            logger.error(f"Impossible to get output_shape for {output.names.pop()}")
+
+        str_to_hash += shape_str + str(output.get_element_type().get_type_name()) + \
+            str(output_node.get_type_info().name) + str(output_node.get_type_info().version_id)
+
+    return str_to_hash
+
+def create_hash(in_dir_path: Path, operations=dict()):
```
Comment (iefode) [inline src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py:113]: split shared functions to separated file

File: src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py (hunk 4 above)
Comment (andrei-kochin) [reply src/tests/test_utils/functional_test_utils/layer_tests_summary/calculate_ir_hash.py:158]: Rename is not needed

**Pull request closed without merging**